pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -1025,6 +1025,32 @@ func (m *kubeGenericRuntimeManager) computePodActions(ctx context.Context, pod *
 		klog.V(2).InfoS("Message for Container of pod", "containerName", container.Name, "containerStatusID", containerStatus.ID, "pod", klog.KObj(pod), "containerMessage", message)
 	}
 
+	// Restart running sidecar containers which have had their definition changed.
+	if utilfeature.DefaultFeatureGate.Enabled(features.SidecarContainers) {
+		var message string
+		var reason containerKillReason
+		for idx, container := range pod.Spec.InitContainers {
+			containerStatus := podStatus.FindContainerStatusByName(container.Name)
+
+			if !types.IsRestartableInitContainer(&container) || containerStatus == nil || containerStatus.State != kubecontainer.ContainerStateRunning {
+				continue
```

Comment:
can you please clarify. If the sidecar container is not running, why we shouldn't start it? Is this condition can even be hit ever?